src/Chainweb/Pact/PactService.hs

```diff
@@ -162,9 +164,30 @@ withPactService ver cid chainwebLogger bhDb pdb sqlenv config act =
                     , _psBlockGasLimit = _pactBlockGasLimit config
                     , _psChainId = cid
                     }
-            !pst = PactServiceState Nothing mempty initialParentHeader P.noSPVSupport
-        runPactServiceM pst pse $ do
+        let !pst = PactServiceState Nothing mempty initialParentHeader P.noSPVSupport
+
+        when (_pactRosettaEnabled config) $ do
+          putStrLn "oh yeah"
+          (earliestBlockHeight, _) <- catch (_cpGetEarliestBlock checkpointer) $ \(e :: SomeException) -> do
+            putStrLn $ show e
+            throwM e
+          putStrLn "koolaid"
+          let gHeight = genesisHeight ver cid
+          when (gHeight /= earliestBlockHeight) $ do
+            let e = RosettaWithoutFullHistory
+                  { _earliestBlockHeight = earliestBlockHeight
+                  , _genesisHeight = gHeight
+                  }
+            let msg = J.object
+                  [ "details" J..= e
+                  , "message" J..= J.text "Your node has Rosetta enabled,\
+                      \ which requires that the full history is available.\
+                      \ However, the full history is not available.\
+                      \ Perhaps you have compacted your node?"
+                  ]
+            onFatalError e (J.encodeText msg)
```

Comment:
This should not use `onFatalError`: We should throw here, and something else should be catching whatever we throw and logging there.